### PR TITLE
Ports the Artist Bench (without porting the artist)

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
## About The Pull Request
dreamseeker_3T5k9bQZjn
dreamseeker_4cbF99SEyA

This ports the artist's workbench with some differences from upstream.

Whereas you originally needed 20 of each material it uses to start using it to begin with, you now only need 5, and it is noted in its description.  I have also taken the time to change around the lore and fluff of the items to have less out of place descriptions. The gun artwork has also been disabled.

The bench has been placed below the autolathe room.
StrongDMM_FENsoauSTG

## Why It's Good For The Game
Emojicracy results were in favor of porting the workbench without actually porting artists. This PR also includes critical updates to autolathe code that will matter later on.

## Changelog
```changelog Toriate
add: Artist's Bench from upstream has been added! Find it below the autolathe room.
del: Oborin Syndrome has been phased out due to lack of supporting lore.
add: Artistic has been added to replace Oborin Syndrome.
```

### CODE REVIEW

File: code/__DEFINES/gamemode.dm [hunk 1]
```diff
@@ -48,7 +48,7 @@
 #define ROLE_MUTINEER "mutineer"
 #define ROLE_COMMANDO "commando"
 #define ROLE_DEATHSQUAD "deathsquad"
-#define ROLE_ACTOR "actor"
```
Comment (Jamini) [inline code/__DEFINES/gamemode.dm:51]: Uhhhhh, Tori? Are we removing Actor without making it an alt-title for club workers?

File: code/datums/outfits/jobs/engineering.dm [hunk 2]
```diff
@@ -16,7 +16,6 @@
 	name = OUTFIT_JOB_NAME("Chief Engineer")
 	head = /obj/item/clothing/head/hardhat/white
 	uniform = /obj/item/clothing/under/rank/exultant
-	head = /obj/item/clothing/head/hardhat/white
```
Comment (Jamini) [inline code/datums/outfits/jobs/engineering.dm:19]: Why?

File: code/datums/setup_option/backgrounds/fate.dm [hunk 3]
```diff
@@ -87,19 +87,28 @@
...
 /datum/category_item/setup_option/background/fate/lowborn
 	name = "Lowborn"
 	desc = "Your origin is at the bottom of the society, be it from slums, underground trash communities, refugees, or any number of neglected communities. \
 			Maybe, unlike everyone else here, this ship is actually an improvement in living conditions for you...."
 	restricted_jobs = list(/datum/job/captain, /datum/job/chaplain, /datum/job/merchant, /datum/job/cmo, /datum/job/rd, /datum/job/ihc)
 	perks = list(PERK_LOWBORN)
 
+
+/////////// OCCULUS EDIT - Artism!
+/datum/category_item/setup_option/background/fate/artistic
```
Comment (Jamini) [inline code/datums/setup_option/backgrounds/fate.dm:108]: Very much like Artistic

File: code/game/machinery/autolathe/artist_bench.dm [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+#define ERR_NOREAGENT "no reagent"
+#define ERR_NOLICENSE "no license"
+#define ERR_PAUSED "paused"
+#define ERR_NOINSIGHT "no insight"
+#define MAX_STAT_VALUE 12
+
+/obj/machinery/autolathe/artist_bench
+	name = "artist's bench"
+	desc = "Insert wood, steel, glass, plasteel, plastic and a bit of your soul to create a beautiful work of art. \
+			Requires a stock of at least 5 sheets of each of the aforementioned materials before you begin sacrificing your soul."	// OCCULUS EDIT - Makes its requirements clear
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:13]: Why are we reducing this from the Eris requirements? Steel, glass, and plastic are super plentiful.

File: code/game/machinery/autolathe/artist_bench.dm [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+	icon_state = "bench"
+	circuit = /obj/item/weapon/electronics/circuitboard/artist_bench
+	have_disk = FALSE
+	have_reagents = FALSE
+	have_recycling = FALSE
+	have_design_selector = FALSE
+	categories = list("Artwork")
+
+	suitable_materials = list(MATERIAL_WOOD, MATERIAL_STEEL, MATERIAL_GLASS, MATERIAL_PLASTEEL, MATERIAL_PLASTIC)
+	var/min_mat = 5 // OCCULUS EDIT - Makes its requirements less dumb
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:24]: Honestly probably should be 20.

File: code/game/machinery/autolathe/artist_bench.dm [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+
+			if("sniper")//From sniper.dm, Arbitrary values
+				R.caliber = CAL_ANTIM
+				R.bulletinsert_sound = 'sound/weapons/guns/interact/rifle_load.ogg'
+				R.fire_sound = 'sound/weapons/guns/fire/sniper_fire.ogg'
+				R.one_hand_penalty = 15 + rand(-3,5) //From sniper.dm, Temporary values
+				R.recoil_buildup = 90 + rand(-10,10)
+
+			if("gyro")//From gyropistol.dm, Arbitrary values
+				R.caliber = CAL_70
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:241]: 0.70  cal ammo still doesn't have sprites at this time.

File: code/game/machinery/autolathe/artist_bench.dm [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+		weight_artwork_gunmod += 8
+	if(ins_used >= 55)
+		weight_artwork_gunmod += 4
+		weight_artwork_tool += 12
+		weight_artwork_toolmod += 12
+	else
+		weight_artwork_statue += 12
+
+	return pickweight(list(
+		//"artwork_revolver" = weight_artwork_revolver,	// OCCULUS EDIT - no meme guns
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:165]: Honestly, I wouldn't mind seeing guns come from this. But that might just be me. Why aren't the toolmods/gunmods and weapon options in here? Is this port finished Eris-side?

File: code/game/machinery/autolathe/artist_bench.dm [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+				)
+		return R
+
+	else if(full_artwork == "artwork_statue")
+		var/obj/structure/artwork_statue/S = new(src)
+		return S
+
+	else if(full_artwork == "artwork_oddity")
+		var/obj/item/weapon/oddity/artwork/O = new(src)
+		var/list/oddity_stats = list(STAT_MEC = rand(0,1), STAT_COG = rand(0,1), STAT_BIO = rand(0,1), STAT_ROB = rand(0,1), STAT_TGH = rand(0,1), STAT_VIG = rand(0,1))//May not be nessecary
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:274]: Okay, so we have a 50% chance of each oddity stat to have +1 in it by default... per stat.

File: code/game/machinery/autolathe/artist_bench.dm [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+		var/stats_amt = 3
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:275]: Then we pick three stats randomly... which CAN be stacked.

File: code/game/machinery/autolathe/artist_bench.dm [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+		if(ins_used >= 85)//Arbitrary values
+			stats_amt += 3
+		if(ins_used >= 70)
+			stats_amt += 3
+		if(ins_used >= 55)
+			stats_amt += 3//max = 3*4*2+6 = 30 points, min 3*4+6 = 18
+		for(var/i in 1 to stats_amt)
+			var/stat = pick(ALL_STATS)
+			oddity_stats[stat] = min(MAX_STAT_VALUE, oddity_stats[stat]+rand(1,2))
```
Comment (Jamini) [inline code/game/machinery/autolathe/artist_bench.dm:284]: Okay so we add 1-2 to a random stat 3-12 times. And we can't go over the MAX_STAT_VALUE on the particular oddity. alright.

**Review by Jamini — DISMISSED** [review]
Please revert your nerf to the required materials Tori. If Eris thinks it should take 20 for this stuff, it probably should take *at least* twenty, if not more of a material.

**Comment by Jamini** [comment]
> Please revert your nerf to the required materials Tori. If Eris thinks it should take 20 for this stuff, it probably should take _at least_ twenty, if not more of a material.

Also please put the white hardhat back on the CE's outfit. Zero reason to remove that.

File: code/__DEFINES/gamemode.dm (hunk 1 above)
Comment (Michiyamenotehifunana) [reply code/__DEFINES/gamemode.dm:51]: Regression that I missed, will undo.

File: code/datums/outfits/jobs/engineering.dm (hunk 2 above)
Comment (Michiyamenotehifunana) [reply code/datums/outfits/jobs/engineering.dm:19]: No idea, wasn't even mentioned in the PR I pulled that from.

File: code/game/machinery/autolathe/artist_bench.dm (hunk 4 above)
Comment (Michiyamenotehifunana) [reply code/game/machinery/autolathe/artist_bench.dm:13]: It's not reduced, that is simply the minimum stock of materials required. What it consumes is independent of that.

File: code/game/machinery/autolathe/artist_bench.dm (hunk 7 above)
Comment (Michiyamenotehifunana) [reply code/game/machinery/autolathe/artist_bench.dm:165]: At present, weapons, tools, and gunmods appear to be unimplemented, and also lacking in sprites.

**Comment by Michiyamenotehifunana** [comment]
Regressions reverted, minimum requirement has remained unchanged as it does not affect the cost, and previously you'd be left unable to create a piece of art if you didn't have 20 sheets of a material the artwork isn't even using.

It has also been mapped in now. 
StrongDMM_FENsoauSTG

**Review by Jamini — APPROVED** [review]
It looks like my concerns were addressed... let's do this.

**Pull request merged**